src/main/db_main.cpp

```diff
@@ -50,10 +50,14 @@ DBMain::DBMain(std::unordered_map<settings::Param, settings::ParamInfo> &&param_
   thread_pool_->Startup();
 
   t_cop_ = new trafficcop::TrafficCop;
-  command_factory_ = new network::PostgresCommandFactory;
-
   connection_handle_factory_ = new network::ConnectionHandleFactory(common::ManagedPointer(t_cop_));
-  provider_ = new network::PostgresProtocolInterpreter::Provider(common::ManagedPointer(command_factory_));
+
```

Comment:
This is probably not the right solution. For instance I don't know why `TerrierServer` needs to take in the providers and command factory from the main. I think the server should bring up those things itself, and DBMain should pass the hot_standby telling it to also bring up the replication event loop. @tli2 do you know why DBMain is the owner of the command factory and provider?

Reply:
dependency injection. Otherwise I need to change server code if I need to change the protocol started.

What you should probably do is to change the injected type from a single provider to a list of providers if you plan to run multiple protocols off of the same network I/O layer